src/api.ts

```diff
@@ -340,10 +340,14 @@ export class SharedCache {
     return SharedCache.instance;
   }
 
-  public getPromptCacheKey(id: string, name: string, version: number): string {
+  public getPromptCacheKey(
+    id?: string,
+    name?: string,
+    version?: number
```

Comment:
When handling only optional params in js it's recommended to use an object because it's not great ux to do:
```ts
instance.getPromptCacheKey(undefined, undefined, version)
// vs
instance.getPromptCacheKey({ version })
```